tests/cephadm/test_cephadm_staggered_upgrade.py

```diff
@@ -0,0 +1,123 @@
+    if CephAdm(node).ceph.health() != "HEALTH_OK":
+        raise StaggeredUpgradeError("Cluster is in HEALTH_ERR state")
+    # Set osd flags
+    for flag in osd_flags:
+        if CephAdm(node).ceph._osd.set(flag):
+            raise StaggeredUpgradeError("Unable to set osd flag")
+    # Check target image
+    if CephAdm(node).ceph.orch.upgrade.check(image=target_image):
+        raise StaggeredUpgradeError("Upgrade image check failed")
+    # Staggered upgrade with daemon_types
+    if action == "daemon_types":
+        daemon_types = config.get("daemon_types")
+        if daemon_types == "osd":
+            limit = config.get("limit")
+            if CephAdm(node).ceph.orch.upgrade.start(
+                image=target_image, daemon_types=daemon_types, limit=limit
+            ):
+                raise StaggeredUpgradeError("Unable to start upgrade with daemon_types")
+        else:
+            if CephAdm(node).ceph.orch.upgrade.start(
+                image=target_image, daemon_types=daemon_types
+            ):
+                raise StaggeredUpgradeError("Unable to start upgrade with daemon_types")
+    # Staggered upgrade with services
+    if action == "services":
+        services = config.get("services")
+        if services == "osd.all_available_devices":
+            limit = config.get("limit")
+            if CephAdm(node).ceph.orch.upgrade.start(
+                image=target_image, services=services, limit=limit
+            ):
+                raise StaggeredUpgradeError("Unable to start upgrade with services")
+        else:
+            if CephAdm(node).ceph.orch.upgrade.start(
+                image=target_image, services=services
+            ):
+                raise StaggeredUpgradeError("Unable to start upgrade with services")
+    # Staggered upgrade with hosts
+    if action == "hosts":
+        nodes = config.get("nodes")
+        hosts = ",".join(
+            [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+        )
+        if CephAdm(node).ceph.orch.upgrade.start(image=target_image, hosts=hosts):
+            raise StaggeredUpgradeError("Unable to start upgrade with hosts")
+    # Staggered upgrade with all combinations
+    if action == "all_combination":
+        nodes = config.get("nodes")
+        daemon_types = config.get("daemon_types")
+        limit = config.get("limit")
+        services = config.get("services")
+        hosts = ",".join(
+            [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+        )
+        if CephAdm(node).ceph.orch.upgrade.start(
+            image=target_image, daemon_types=daemon_types, limit=limit, hosts=hosts
+        ):
+            raise StaggeredUpgradeError("Unable to start upgrade with all combinations")
+    # Check upgrade status
+    check_upgrade_status(node)
+    # Check cluster health after upgrade
+    if orch.cluster.check_health(rhbuild=config.get("rhbuild"), client=orch.installer):
```

Comment:
Need to update here as well @MohitBis . Sorry I missed in the previous comment. 
So we can remove the orch class object as well, if not used anywhere else

Reply:
Thanks @pranavprakash20 
Updated